tests/instruments/test_all_instruments.py

```diff
@@ -49,6 +49,17 @@
             if b:
                 devices.append(d)
 
+
+# Collect all properties
+# TODO add Channel properties as well: How collect all Channel classes?
+properties = []
+for device in devices:
+    for property_name in dir(device):
+        prop = getattr(device, property_name)
+        if isinstance(prop, property):
+            properties.append((device, property_name, prop))
```

Comment:
Does that result in "nice" test ids in the pytest parametrisation? (Visible when a test fails) Otherwise using a namedtuple might help.

Reply:
> Does that result in "nice" test ids in the pytest parametrisation? (Visible when a test fails) Otherwise using a namedtuple might help.

I use the values (class and property name) to create a meaningful error message in the test (after `assert`).

Yes, test ids were meaningless, but I used a different option to set the test_ids according to https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids

Reply:
Nice!